### PR TITLE
Reworked server select in anime.py

### PR DESCRIPTION
**Pull request by Blatzar**
Couldn't find a good way to place it anywhere else you requested. Making this draft so we could get this done.

https://github.com/vn-ki/anime-downloader/pull/360#issuecomment-623716418 for more info

### CODE REVIEW

File: anime_downloader/sites/anime.py [hunk 1]
```diff
@@ -352,6 +352,50 @@ def get_data(self):
...
+        as it prioritizes preferred language over preferred server
+        """
+
+        version = self.config.get('version','subbed') #TODO add a flag for this
+        servers = self.config.get('servers',[''])
+
+        logger.debug('Data : {}'.format(data))
+
+        sources = []
+        for a in servers: #Servers in order in correct language
```
Comment (vn-ki) [inline anime_downloader/sites/anime.py:386]: Don;t use `a` and `b`

File: anime_downloader/sites/anime.py [hunk 2]
```diff
@@ -352,6 +352,50 @@ def get_data(self):
...
+            for b in (list(filter(lambda server: (server['version'] == version and server['config'] == a and get_extractor(server['extractor']) != None) , data))):
```
Comment (vn-ki) [inline anime_downloader/sites/anime.py:387]: Make the check a different function.

File: anime_downloader/sites/anime.py (hunk 1 above)
Comment (Blatzar) [reply anime_downloader/sites/anime.py:386]: Crap I have to learn real programming conventions

File: anime_downloader/sites/anime.py [hunk 3]
```diff
@@ -352,6 +352,50 @@ def get_data(self):
...
+                sources.append(b)
+
+        for a in servers: #Servers in order in incorrect language
+            for b in (list(filter(lambda server: (server['version'] != version and server['config'] == a and get_extractor(server['extractor']) != None) , data))):
+                sources.append(b)
+
+        logger.debug('Sorted sources : {}'.format(sources))
+
+        return '' if not sources else [(sources[0]['extractor'],sources[0]['url'])]
```
Comment (vn-ki) [inline anime_downloader/sites/anime.py:396]: Shouldn't you be returning empty list?

Comment (Blatzar) [reply anime_downloader/sites/anime.py:396]: Ideally yes, but to get the no sites found error you gotta return nothing iirc, following this.

Comment (vn-ki) [reply anime_downloader/sites/anime.py:396]: Then return `None`

File: anime_downloader/sites/anime.py [hunk 4]
```diff
@@ -352,6 +352,49 @@ def get_data(self):
...
+
+        version = self.config.get('version','subbed') #TODO add a flag for this
+        servers = self.config.get('servers',[''])
+
+        logger.debug('Data : {}'.format(data))
+
+        """Sorts the dicts by preferred server in config"""
+        sorted_by_server = sorted(data, key=lambda x: servers.index(x['server']))
+        """
+        Sorts the above by preferred language
```
Comment (vn-ki) [inline anime_downloader/sites/anime.py:388]: Use `#` comments for code comments.

File: anime_downloader/sites/anime.py [hunk 5]
```diff
@@ -352,6 +352,49 @@ def get_data(self):
...
+        logger.debug('Data : {}'.format(data))
+
+        """Sorts the dicts by preferred server in config"""
+        sorted_by_server = sorted(data, key=lambda x: servers.index(x['server']))
+        """
+        Sorts the above by preferred language 
+        resulting in a list with the dicts sorted by language and server
+        with language being prioritized over server
+        """
+        sorted_by_lang = sorted(sorted_by_server, key=lambda x: x['version'] == version, reverse=True)
```
Comment (vn-ki) [inline anime_downloader/sites/anime.py:392]: Convert it into a list

File: anime_downloader/sites/anime.py (hunk 4 above)
Comment (Blatzar) [reply anime_downloader/sites/anime.py:388]: Oh I thought docstrings for everything since you reviewed selescrape and then started using it, will fix

**Pull request merged**